### ISSUE TITLE
Export lem-base::*current-buffer*?

### ISSUE THREAD

**Issue by kchanqvq** [lem-project/lem#1152]
What is the current way to temporarily change `(current-buffer)` (like `with-current-buffer` in Emacs)?  Although setter to `current-buffer` is exported, implementing `with-current-buffer` using the setter is not thread-safe. The best way seems to be just bind `*current-buffer*`, however currently it is internal to `lem-base` which suggest it is only used for internal implementation.   Can we export this symbol instead?

**Comment by cxxxr** [comment]
I believe that active use of dynamic scopes should be avoided except in a few cases, such as testing.
I don't think dynamic scope is necessary since the operator provided by lem explicitly specifies the arguments, but what kind of case is there?

**Comment by kchanqvq** [comment]
I agree with you from language design view point, that one should pass buffer as argument (conveniently packaged in point objects) in favor of `with-current-buffer` and `current-buffer`. Being said that, lem already provides and exports `current-buffer` which seems mostly useful to the coding style that relies on current buffer as a dynamic context. Then it might as well export something equivalent to `with-current-buffer`.

The concrete use case is that I'm writing a compatibility layer to run Emacs Lisp directly on lem. I have got surprisingly far! Lem design seems to follow very closely to that of Emacs. (I'm not familiar with the design process of lem, is this intended?) I've ported quite a few Emacs functions relying on `current-buffer`, and I also need to port `with-current-buffer`. I can do still do that if `lem-base::*current-buffer*` is not exported -- I just need to maintain my own `el-compat:*current-buffer*` instead, but I find this awkward given that there's already `lem-base:current-buffer` and `lem-base::*current-buffer*`.
